### ISSUE TITLE
Support for Brotli compression on Cloudfront 

### ISSUE THREAD

**Issue by edhgoose** [hashicorp/terraform-provider-aws#15172]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS have announced support for Brotli on Cloudfront: https://aws.amazon.com/about-aws/whats-new/2020/09/cloudfront-brotli-compression/ which in theory enables smaller file sizes, and is widely supported by Chrome and Firefox.

It would be great to be able to enable this across our CloudFront distributions.

### New or Affected Resource(s)

* aws_cloudfront_distribution

### Potential Terraform Configuration

Currently the default_cache_behavior just has `compress` with a boolean value:

```hcl
  default_cache_behavior {
    compress = true
}
```

Perhaps:

```hcl
  default_cache_behavior {
    compress = ['gzip', 'br']
}
```

would be better, if possible, to enable users to make a decision on which compression types to support? I'd personally be happy with just `compress = true` supporting both gzip and br, but I'm not sure if that's universal and would change existing distributions.

### References

* https://aws.amazon.com/about-aws/whats-new/2020/09/cloudfront-brotli-compression/
* https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/ServingCompressedFiles.html

**Comment by ewbankkit** [comment]
Requires [AWS SDK v1.34.21](https://github.com/aws/aws-sdk-go/releases/tag/v1.34.21):

* https://github.com/terraform-providers/terraform-provider-aws/pull/14955

Looking at the [_Working with policies_](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/controlling-the-cache-key.html#cache-policy-compressed-objects) section of the CloudFront Developer Guide, Brotli compression is enabled via an attribute on the Cache Policy object: https://github.com/terraform-providers/terraform-provider-aws/issues/14373, https://github.com/terraform-providers/terraform-provider-aws/pull/14672.
On the `aws_cloudfront_distribution` resource, `compress` would stay as a boolean in the various `cache_behavior` configuration blocks.

**Comment by roberth-k** [comment]
Now included in https://github.com/hashicorp/terraform-provider-aws/pull/14672 as `enable_accept_encoding_brotli`.

**Comment by ewbankkit** [comment]
This has been included in the new `aws_cloudfront_cache_policy` resource implemented via https://github.com/hashicorp/terraform-provider-aws/pull/17336, to be released in **v3.28.0** of the Terraform AWS Provider
